Short circuit if all checks disabled

### Bug description

Running "pylint test.py --disable=all" takes more than 3s!
```sh
$ touch test.py
$ time pylint test.py --disable=all

real    0m3.684s
user    0m0.000s
sys     0m0.015s
```
Running pylint without "disable=all" on a little project (150-lines telegram bot) takes more than 8s. It is non-usable.
```sh
$ time pylint main.py
************* Module main
main.py:137:7: R0133: Comparison between constants: '0 == 1' has a constant value (comparison-of-constants)
main.py:147:0: C0116: Missing function or method docstring (missing-function-docstring)

------------------------------------------------------------------
Your code has been rated at 9.57/10 (previous run: 9.57/10, +0.00)


real    0m8.352s
user    0m0.000s
sys     0m0.000s
```

### Configuration

_No response_

### Command used

```shell
pylint test.py
```


### Pylint output

```shell
...
```


### Expected behavior

It is unacceptable that even on an empty file pylint runs for at least 3 seconds. I use the VS Code extension in my project, which for a small example with a 150-line project reacts to changes in 8 (!) seconds. This is literally impossible to use.

### Pylint version

```shell
$ pylint --version
pylint 2.17.4
astroid 2.15.5
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
```


### OS / Environment

Windows 11

### Additional dependencies

_No response_

The gpt4free repository was cloned in this project, I added it to ignore in the .pylintrc file. This reduced the running time by only 2 seconds.
Checking the file with "\n":
```sh
$ echo "\n" > test.py && time pylint test.py
************* Module test
test.py:1:2: E0001: Parsing failed: 'unexpected character after line continuation character (<unknown>, line 1)' (syntax-error)

real    0m1.639s
user    0m0.000s
sys     0m0.015s
```
Checking the bot file (150 lines):
```sh
$time pylint main.py
************* Module main
main.py:137:7: R0133: Comparison between constants: '0 == 1' has a constant value (comparison-of-constants)
main.py:147:0: C0116: Missing function or method docstring (missing-function-docstring)

------------------------------------------------------------------
Your code has been rated at 9.57/10 (previous run: 9.57/10, +0.00)


real    0m6.689s
user    0m0.000s
sys     0m0.031s
```
There are no big files in the working directory.
I created an empty file in empty folder, here is the output of pylint:
```sh
$ mkdir pylint-testing && cd pylint-testing

$ touch test.py

$ time pylint test.py --disable=all

real    0m1.616s
user    0m0.000s
sys     0m0.015s

$ time pylint test.py

real    0m1.592s
user    0m0.000s
sys     0m0.000s
```
I don't think it's a problem of specific libraries. If you want me to share any logs or traces with you, let me know.

--- a/doc/whatsnew/fragments/8715.performance
+++ b/doc/whatsnew/fragments/8715.performance
@@ -1,0 +1,3 @@
+Exit immediately if all messages are disabled.
+
+Closes #8715

--- a/pylint/lint/run.py
+++ b/pylint/lint/run.py
@@ -175,10 +175,13 @@ group are mutually exclusive.",
                 sys.exit(code)
             return
 
-        # Display help messages if there are no files to lint
-        if not args:
+        # Display help if there are no files to lint or no checks enabled
+        if not args or len(linter.config.disable) == len(
+            linter.msgs_store._messages_definitions
+        ):
             print(linter.help())
-            sys.exit(32)
+            exit_code = 32 if not args else 0
+            sys.exit(exit_code)
 
         if linter.config.jobs < 0:
             print(

--- a/pylint/lint/run.py
+++ b/pylint/lint/run.py
@@ -179,9 +179,8 @@ group are mutually exclusive.",
         if not args or len(linter.config.disable) == len(
             linter.msgs_store._messages_definitions
         ):
-            print(linter.help())
-            exit_code = 32 if not args else 0
-            sys.exit(exit_code)
+            print("No files to lint: exiting.")
+            sys.exit(32)
 
         if linter.config.jobs < 0:
             print(

--- a/tests/test_self.py
+++ b/tests/test_self.py
@@ -214,8 +214,8 @@ class TestRunTC:
 
     def test_disable_all(self) -> None:
         out = StringIO()
-        self._runtest([UNNECESSARY_LAMBDA, "--disable=all"], out=out, code=0)
-        assert "show this help message and exit" in out.getvalue().strip()
+        self._runtest([UNNECESSARY_LAMBDA, "--disable=all"], out=out, code=32)
+        assert "No files to lint: exiting." in out.getvalue().strip()
 
     def test_no_out_encoding(self) -> None:
         """Test redirection of stdout with non ascii characters."""

--- a/tests/test_self.py
+++ b/tests/test_self.py
@@ -212,6 +212,11 @@ class TestRunTC:
     def test_error_missing_arguments(self) -> None:
         self._runtest([], code=32)
 
+    def test_disable_all(self) -> None:
+        out = StringIO()
+        self._runtest([UNNECESSARY_LAMBDA, "--disable=all"], out=out, code=0)
+        assert "show this help message and exit" in out.getvalue().strip()
+
     def test_no_out_encoding(self) -> None:
         """Test redirection of stdout with non ascii characters."""
         # This test reproduces bug #48066 ; it happens when stdout is redirected

